### DISCUSSION TITLE
updates and possibly new bugs discovered

### DISCUSSION THREAD

**Discussion by chengidesk** [netalertx/NetAlertX#804] [General]
Hi Jokob-sk,

Happy to shared that by utilizing your system, we registered 320 devices (hardware MAC) from students for a school back in mid Aug. We also discovered there were 260 unknown client (some are due to the random MAC).

Noticed possible bugs when I click on connected devices, some of it are grey

![NetAlertX-1](https://github.com/user-attachments/assets/ce0f12e1-e0a4-4a8e-bc19-cb818d1d7c1d)
![NetAlertX-2](https://github.com/user-attachments/assets/15012f0d-cf1e-4bd5-9360-4838ab100cec)

Thank you for your assistant.

Regards,
Cheng-I

**Comment by jokob-sk** [comment]
Hi @chengidesk ,

Glad to hear that! I think that the incorrect color/label was a bug in one of the older versions. Can you please try to update to the latest version if you are not already using it?

**Comment by chengidesk** [comment]
Hi @jokob-sk,

I just recreated the container, but it appear the same

![image](https://github.com/user-attachments/assets/f56c092b-3222-4c50-af16-47bfb1cc6e8c)
![image](https://github.com/user-attachments/assets/6cafa093-08f0-4d4c-8fa5-e27145466d62)

Thank and regards,

**Reply by jokob-sk** [reply]
Hi @chengidesk ,

The latest version is `24.9.12`: https://github.com/jokob-sk/NetAlertX/releases/tag/v24.9.12

You usually need to stop, remove the container, the image and recreate it, or force pull (`--pull always`) the new image. For example:

```
sudo docker run --pull always --rm --network=host
-v /app/config:/app/config
-v /app/db:/app/db
-v /app/log:/app/log
-e TZ=America/New_York
-e PORT=20211
jokobsk/netalertx
```

**Reply by chengidesk** [reply]
i tried the latest version but it was automatically updated by portainer.
with NetAlertX
![image](https://github.com/user-attachments/assets/5c59c9a7-36b0-4fab-8168-cafd541face7)
without NetAlertX
![image](https://github.com/user-attachments/assets/e56b2f97-7397-48c6-b173-00420068811e)

Current version:
![image](https://github.com/user-attachments/assets/7019b533-e281-4777-809c-deb5aa28de9a)

Wondering how do revert to the previous version ?

Thanks @jokob-sk

**Reply by jokob-sk** [reply]
Please double check your database size and if the maintenance tasks are running (increase their timeout if needed). Usually high resource consumption is linked to a big database, big log files and not executing maintenance tasks. 

Please run the maintenance and DB cleanup plugins with an increased timeout if needed:
![image](https://github.com/user-attachments/assets/ac7f4cb2-08eb-430e-9fa4-143b0b407b6f)

**Reply by chengidesk** [reply]
i tried it does have improvement but the utilization is still high intermittent, any idea? 
![image](https://github.com/user-attachments/assets/41509cfe-684c-443e-9003-ee9418b459ea)

I also notice the main page 
![image](https://github.com/user-attachments/assets/73709b36-9c7d-4ad4-b190-e237f4d843c8)

**Reply by jokob-sk** [reply]
Could you please let me know the size of the database and the log file? You can find it under the Maintenance section.

**Reply by chengidesk** [reply]
Hope these are what you are looking for

![image](https://github.com/user-attachments/assets/19badf40-9c7f-484c-be91-91e292b431c6)
[app.log](https://github.com/user-attachments/files/17226629/app.log)

**Reply by jokob-sk** [reply]
Yes, the DB size is too big I think. Try running the DB clean up task and double check if it's finishing without an error.

Decrease DAYS_TO_KEEP_EVENTS, WORKFLOWS_AppEvents_hist, MAINT_LOG_LENGTH and run the DB clean up and maintenance plugins. Make sure they finish without an error.

Lastly restart the container.

**Reply by chengidesk** [reply]
i've done the above but it is still intermittent. is there a way i could revert back to earlier version and test again?

**Reply by jokob-sk** [reply]
You can specify the image tag in the docker compose file:

```
version: "3"
services:
    web:
        image: my-image:v3
```

From my experience only cleaning the database and shrinking it will help.

You can also export the devices.csv file and the app.conf file and start with a clean DB

**Reply by chengidesk** [reply]
i deleted the whole container, db and reinstall but i am not able to restore the devices.csv file or maybe i don't know how to restore it... now the memory and cpu utilization is back to expected.

**Reply by jokob-sk** [reply]
You should be able to do it under Maintenance:
![image](https://github.com/user-attachments/assets/91c1722d-ad17-459d-963e-e93c25315fd8)

More info here: https://github.com/jokob-sk/NetAlertX/blob/main/docs/BACKUPS.md

**Reply by chengidesk** [reply]
Thanks @jokob-sk ;
I managed to restore devices.csv. Apperently the db file size grew again but there is no spike in cpu/memory usage. Apps logs gre to 126,30mb

![image](https://github.com/user-attachments/assets/75d258f7-3eb4-43fe-bb0b-b5d576bb895c)

**Reply by chengidesk** [reply]
Another question is how do i get back this colorful count when I click on devices like monitoring? 
Currently it is only showing 
![image](https://github.com/user-attachments/assets/9e8ff3b2-b6aa-4e15-841e-c80874a061fa)

Under monitoring
![image](https://github.com/user-attachments/assets/3a79e906-957a-430d-9364-2764c6b6e6a1)

**Reply by jokob-sk** [reply]
glad to hear that, you can also lower the LOG_LEVEL to e.g. `minimal` if you have a higher level set currently. You can also try to decrease the `PLUGINS_KEEP_HIST` setting e.g. 100 ... and `WORKFLOWS_AppEvents_hist` to e.g. 10000

it's weird that the Plugins_History, AppEvents are that many - can you double check the DB Cleanup plugin is not failing?

**Reply by jokob-sk** [reply]
You can select which cards a shown here: 
![image](https://github.com/user-attachments/assets/1d5e3e0c-1b09-40fa-8cf6-7133be354a8c)

**Reply by jokob-sk** [reply]
I wrote up this short PERFORMANCE guide based on this thread: https://github.com/jokob-sk/NetAlertX/blob/main/docs/PERFORMANCE.md

**Reply by chengidesk** [reply]
@jokob-sk thank you very much. Now more or less on UI everything back to normal :)
changed 
PLUGINS_KEEP_HIST to 100
WORKFLOWS_AppEvents_hist already showing 5000.

I'll come back with the others checking later. thanks for the assistant !
